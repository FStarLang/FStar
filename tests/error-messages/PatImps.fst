module PatImps

let f1 (x:list 'a) : int =
  match x with
  | Nil -> 0
  | Cons _ _ -> 1

noeq
type t2 = | A : #x:int -> t2

let f2 (x:t2) : int =
  match x with
  | A #i -> i

noeq
type t3 =
  | B : #x:int -> t3
  | C : #x:int -> t3

let f3 (x:t3) : int =
  match x with
  | B #i -> i
  | C #i -> i
