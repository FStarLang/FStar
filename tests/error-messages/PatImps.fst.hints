[
  "‡zê\u0016©\u000f~¥›&Æ­\u001f\u000b‚B",
  [
    [
      "PatImps.f1",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query", "disc_equation_Prims.Cons",
        "disc_equation_Prims.Nil", "fuel_guarded_inversion_Prims.list",
        "projection_inverse_BoxBool_proj_0"
      ],
      0,
      "3d92d015726030d3cbed8a1d13601aba"
    ],
    [
      "PatImps.__proj__B__item__x",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_d12b69dbfc634a024ebe6635710ee639"
      ],
      0,
      "09531fdf521634c80ebf99aab9602296"
    ],
    [
      "PatImps.__proj__C__item__x",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "refinement_interpretation_Tm_refine_d42021b5d15fa2ad400d571ee4741f6a"
      ],
      0,
      "9b5ef41627b3868a27cccb35c43ab37c"
    ],
    [
      "PatImps.f3",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query", "disc_equation_PatImps.B",
        "disc_equation_PatImps.C", "fuel_guarded_inversion_PatImps.t3",
        "projection_inverse_BoxBool_proj_0"
      ],
      0,
      "eb3ea7a03000d8f3ee982aa20b08c8e3"
    ]
  ]
]