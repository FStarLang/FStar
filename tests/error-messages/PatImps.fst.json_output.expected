Module after desugaring:
module PatImps
Declarations: [
let f1 x =
  (match x with
    | [] -> 0
    | _ :: _ -> 1)
  <:
  Prims.int
noeq
type t2 = | A : PatImps.t2


let f2 x =
  (let PatImps.A #i = x in
    i)
  <:
  Prims.int
noeq
type t3 =
  | B : PatImps.t3
  | C : PatImps.t3




let f3 x =
  (match x with
    | PatImps.B #i -> i
    | PatImps.C #i -> i)
  <:
  Prims.int
]

Module before type checking:
module PatImps
Declarations: [
let f1 x =
  (match x with
    | [] -> 0
    | _ :: _ -> 1)
  <:
  Prims.int
noeq
type t2 = | A : PatImps.t2


let f2 x =
  (let PatImps.A #i = x in
    i)
  <:
  Prims.int
noeq
type t3 =
  | B : PatImps.t3
  | C : PatImps.t3




let f3 x =
  (match x with
    | PatImps.B #i -> i
    | PatImps.C #i -> i)
  <:
  Prims.int
]

Module after type checking:
module PatImps
Declarations: [
let f1 x =
  (match x with
    | [] -> 0
    | _ :: _ -> 1)
  <:
  Prims.int
noeq
type t2 = | A : PatImps.t2




let f2 x =
  (let PatImps.A #i = x in
    i)
  <:
  Prims.int
noeq
type t3 =
  | B : PatImps.t3
  | C : PatImps.t3








let f3 x =
  (match x with
    | PatImps.B #i -> i
    | PatImps.C #i -> i)
  <:
  Prims.int
]

